smalltalksrc/Slang/TMethod.class.st

```diff
@@ -1440,24 +1466,27 @@ TMethod >> inlineFunctionCall: aSendNode in: aCodeGen [
 	doNotRename := Set withAll: args.
 	argsForInlining := aSendNode argumentsForInliningCodeGenerator:
 		                   aCodeGen.
-	meth args with: argsForInlining do: [ :argName :exprNode | 
+	meth args with: argsForInlining do: [ :argName :exprNode |
 		exprNode isLeaf ifTrue: [ doNotRename add: argName ] ].
-	(meth statements size = 2 and: [ 
-		 meth statements first isSend and: [ 
-			 meth statements first selector == #flag: ] ]) ifTrue: [ 
+	(meth statements size = 2 and: [
+		 meth statements first isSend and: [
+			 meth statements first selector == #flag: ] ]) ifTrue: [
 		meth statements removeFirst ].
 	meth renameVarsForInliningInto: self except: doNotRename in: aCodeGen.
 	meth renameLabelsForInliningInto: self.
 	self addVarsDeclarationsAndLabelsOf: meth except: doNotRename.
 	substitutionDict := Dictionary new: meth args size * 2.
-	meth args with: argsForInlining do: [ :argName :exprNode | 
+	meth args with: argsForInlining do: [ :argName :exprNode |
 		substitutionDict at: argName put: exprNode.
-		(doNotRename includes: argName) ifFalse: [ 
+		(doNotRename includes: argName) ifFalse: [
 			self removeLocal: argName ] ].
 	meth parseTree bindVariablesIn: substitutionDict.
-	^ meth parseTree endsWithReturn
-		  ifTrue: [ meth parseTree copyWithoutReturn ]
-		  ifFalse: [ meth parseTree ]
+	meth parseTree endsWithReturn ifTrue: [
+		meth parseTree: meth parseTree copyWithoutReturn ].
+	(self parentWontUse: aSendNode) ifTrue: [
+		meth parseTree parent: aSendNode parent.
```

Comment:
Wait, this is strange. You are setting telling a method to have a parent node. I understand you use it below, but this seems like pushing the limits of the model and it can have unexpected effects. Isn't there another way to solve it?